src/util/color.ts

```diff
@@ -126,3 +126,19 @@ export function getTitleAttr(bucket: IBucket, e: IEvent) {
     return e.data.title;
   }
 }
+
+export function getCategoryColorFromEvent(bucket: IBucket, e: IEvent) {
+  if (bucket.type == 'currentwindow') {
+    return getCategoryColorFromString(e.data.app + ' ' + e.data.title);
+  } else if (bucket.type == 'web.tab.current') {
+    return getCategoryColorFromString(e.data.title + ' ' + e.data.url);
```

Comment:
The only issue I can find is that this will possibly not work with some category rules which reference `$` (start of line/content) or `^` (end of line/content).
```suggestion
    // NOTE: this will not work/break category rules which reference `$` or `^`
    return getCategoryColorFromString(e.data.app + ' ' + e.data.title);
  } else if (bucket.type == 'web.tab.current') {
    // NOTE: same as above
    return getCategoryColorFromString(e.data.title + ' ' + e.data.url);
```